### PR TITLE
Remove log shipper plugin

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

Remove unused plugin that will be replaced by direct log sending

## Checklist

- [x] This is something else
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

**Review by JonasKunz — APPROVED** [review]

File: apm-agent-plugins/apm-log-shipper-plugin/src/main/java/co/elastic/apm/agent/log/shipper/LogShipperFactory.java [hunk 1]
```diff
@@ -1,86 +0,0 @@
...
-        ApmServerLogShipper logShipper = new ApmServerLogShipper(apmServerClient, tracer.getConfig(ReporterConfiguration.class), serializer);
-        ExecutorUtils.SingleNamedThreadFactory threadFactory = new ExecutorUtils.SingleNamedThreadFactory(ThreadUtils.addElasticApmThreadPrefix("log-shipper"));
-        fileTailer = new FileTailer(logShipper, BUFFER_SIZE, MAX_LINES_PER_CYCLE, IDLE_TIME_MS, threadFactory);
-    }
-
-    @Override
-    public void start(ElasticApmTracer tracer) throws IOException {
-        LoggingConfiguration config = tracer.getConfig(LoggingConfiguration.class);
-        String logFile = config.getLogFile();
-        if (!config.isShipAgentLogs()) {
```
[the remark this answers is not in the document]
Comment (SylvainJuge) [reply apm-agent-plugins/apm-log-shipper-plugin/src/main/java/co/elastic/apm/agent/log/shipper/LogShipperFactory.java:61]: This is currently being used in https://github.com/elastic/apm-agent-java/pull/2694, but we might have to change the naming of the option when aligning accross agents.

[the remark this answers is not in the document]
Comment (felixbarny) [reply apm-agent-plugins/apm-log-shipper-plugin/src/main/java/co/elastic/apm/agent/log/shipper/LogShipperFactory.java:61]: IIRC, that PR is using a different option. So we can probably remove this option.

[the remark this answers is not in the document]
Comment (SylvainJuge) [reply apm-agent-plugins/apm-log-shipper-plugin/src/main/java/co/elastic/apm/agent/log/shipper/LogShipperFactory.java:61]: My bad, this one is `ship_agent_logs` and the draft PR uses `ship_logs` :man_facepalming: , +1 to remove it then, there is no need to make any distinction between app and agent logs and we should send all of them.

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -26,6 +26,10 @@ endif::[]
 [float]
 ===== Breaking changes
 
+[float]
+===== Refactorings
+* Remove unused 'log-shipper-plugin' and related internal config options - {pull}2848[#2848]
+
```
Comment (felixbarny) [inline CHANGELOG.asciidoc:32]: As this has no user facing impact, I'd just remove this from the release notes.

```suggestion
```

**Pull request merged**